adhocracy4/polls/assets/PollQuestion.jsx

```diff
@@ -1,124 +1,72 @@
+            if (!props.question.multiple_choice) {
+              return (
+                <label className="poll-row radio" key={choice.id} htmlFor={'id_choice-' + choice.id + '-single'}>
+                  <input
+                    className="poll-row__radio radio__input"
+                    type="radio"
+                    name="question"
+                    id={'id_choice-' + choice.id + '-single'}
+                    value={choice.id}
+                    checked={checked}
+                    onChange={(e) => { props.OnSingleChange(props.question.id, e.target.value).bind(this) }}
+                    disabled={!props.question.authenticated}
+                  />
+                  <span className="radio__text">{choice.label}</span>
+                  {choice.is_other_choice &&
                     <input
-                      className="poll-row__radio radio__input"
-                      type="radio"
+                      className="input-group__input"
+                      type="text"
                       name="question"
                       id={'id_choice-' + choice.id + '-single'}
-                      value={choice.id}
-                      checked={checked}
-                      onChange={this.handleOnChange.bind(this)}
-                      disabled={!this.state.question.authenticated}
-                    />
-                    <span className="radio__text">{choice.label}</span>
-                    {choice.is_other_choice &&
-                      <input
-                        className="input-group__input"
-                        type="text"
-                        name="question"
-                        id={'id_choice-' + choice.id + '-single'}
-                        onChange={this.handleTextChange.bind(this)}
-                        disabled={!this.state.question.authenticated}
-                      />}
-                  </label>
-                )
-              } else {
-                return (
-                  <label className="poll-row radio" key={choice.id} htmlFor={'id_choice-' + choice.id + '-multiple'}>
+                      onChange={(e) => { props.onOtherChange(props.question.id, e.target.value) }}
+                      disabled={!props.question.authenticated}
+                    />}
+                </label>
+              )
+            } else {
+              return (
+                <label className="poll-row radio" key={choice.id} htmlFor={'id_choice-' + choice.id + '-multiple'}>
+                  <input
+                    className="poll-row__radio radio__input"
+                    type="checkbox"
+                    name="question"
+                    id={'id_choice-' + choice.id + '-multiple'}
+                    value={choice.id}
+                    checked={checked}
+                    onChange={(e) => { props.OnMultiChange(props.question.id, e.target.value).bind(this) }}
```

Comment:
think you can omit `.bind(this)` and a little typo `OnMulti..` should be `onMulti..` right?